### PR TITLE
Update `tes3.rayTest` documentation

### PR DESCRIPTION
**Pull request by herbertgithubaccount**
Added more detail to the function description and its parameters. Also reorganized the `tip` optimization suggestions.

### CODE REVIEW

File: autocomplete/definitions/global/tes3/rayTest.lua [hunk 1]
```diff
@@ -1,54 +1,111 @@
...
+Here is an overview of how some commonly used parameters will alter how `tes3.rayTest` checks for collisions:
+	1. `root`: Things that aren't a `child` of the specified `root` will be skipped. (If `root` is not provided, then nothing will be skipped by this process.)
 
-!!! tip Improving performance of rayTest
+	2. `ignore`: objects in this array will be skipped.
+
+	3. `maxDistance`: if specified, only objects within the specified distance will be checked.
 
-		1. Keep maximum size of objects reasonable, as well as triangle counts
+	4. `findAll`: If `true`, then all intersections will be returned. Otherwise, only the first intersection will be returned.
```
Comment (C3pa) [inline autocomplete/definitions/global/tes3/rayTest.lua]: Small nit, please make it consistent, so each sentence after the colon starts in either capital or lower case.

File: autocomplete/definitions/global/tes3/rayTest.lua [hunk 2]
```diff
@@ -1,54 +1,111 @@
 return {
 	type = "function",
-	description = [[Performs a ray test and returns various information related to the result(s). If `findAll` is set, the result will be a table of results, otherwise only the first result is returned.
+	description = [[Performs a ray test and returns various information related to the result(s). The ray test works by effectively shooting out a line, starting at `position` and pointing towards `direction`, and then checking to see which objects intersect that line.
+	
+Here is an overview of how some commonly used parameters will alter how `tes3.rayTest` checks for collisions:
+	1. `root`: Things that aren't a `child` of the specified `root` will be skipped. (If `root` is not provided, then nothing will be skipped by this process.)
```
Comment (C3pa) [inline autocomplete/definitions/global/tes3/rayTest.lua]: Please remove the brackets.

File: autocomplete/definitions/global/tes3/rayTest.lua [hunk 3]
```diff
@@ -1,54 +1,111 @@
...
+
+	3. `maxDistance`: if specified, only objects within the specified distance will be checked.
 
-		1. Keep maximum size of objects reasonable, as well as triangle counts
+	4. `findAll`: If `true`, then all intersections will be returned. Otherwise, only the first intersection will be returned.
 
-		2. Whenever possible set a maxDistance in your rayTest calls
+!!! tip Improving performance of rayTest
+	You should always try to follow these suggestions as best you can, as they can noticeably improve the performance of this function. 
+	(Although, some of these are fine to ignore if you're calling `tes3.rayTest` fairly infrequently. For example, you can probably get away with skipping the `root` parameter if `tes3.rayTest` is only used in a callback in the `activate` event.)
```
Comment (C3pa) [inline autocomplete/definitions/global/tes3/rayTest.lua:14]: IMHO, the first sentence sounds a bit condescending. Could you make the tone a bit more neutral? Also, it's fine to remove the brackets from the second sentence.

**Comment by herbertgithubaccount** [comment]
i tried to implement your suggestions and remove some of the unnecessary brackets. let me know if i misunderstood/missed anything.

Looking back on it with some fresh eyes, I do agree that the first sentence of the "performance suggestions tip" was a little condescending. Good catch! I tried to reword it to make it more neutral, but would be happy to make further changes. I'm not always aware of how the tone of what i write comes across, and it's nice to know when/how that could be improved. I also removed the "`(Although, some of these are fine to ignore ...`" part of the suggestion because I think it also might've been a little redundant/condescending, but let me know if I should add it back/replace it with something else.
